lib/ecto/query/builder/join.ex

```diff
@@ -229,6 +231,26 @@ defmodule Ecto.Query.Builder.Join do
     end
   end
 
+  defp ensure_on(nil, join) do
```

Comment:
We don't have to warn on cross joins so can do something like this the below. Sorry being a bit lazy so not completely sure if this is the right thing to warn on.

`defp ensure_on(nil, %{qual: qual}) when qual in [:cross_join, :cross_lateral_join], do: true`

Reply:
Great point I completely missed that 🙇
Fixed in https://github.com/elixir-ecto/ecto/pull/4074/commits/de747ba0a890d6f4f64bb9a8d67484366cf5c2d8